### DISCUSSION TITLE
Add the newsletter form on getdoks.org to the Doks template?

### DISCUSSION THREAD

**Discussion by cameronelliott** [thuliteio/doks#396] [Ideas]
Henk,
I'd like to add the newsletter form to my fork of the Doks template.
I can probably struggle through that on my own, as a back-end non-JS developer.

Have you considered adding the newsletter form to the main  Doks template?
If you did that, I would be happy to wire up the JS to get it to work with Google Forms.
I would say one advantage Google Forms (Sheets) has over using Netlify on the backend,
is that at least two different gmail-based mass emailer companies support input directly from Google Sheets.

So if the Doks template could go direct to Google Sheets, then users can use the 
[GMass](https://www.gmass.co/) or [MailMeteor](https://mailmeteor.com/) offerings to send their newsletter with very very high deliverability using their gmail account. (Those two services read directly from GSheets)

I don't really have the Hugo/HTML/CSS skills to submit a PR for the front end work myself.

Anyway, just a wild idea. Maybe it doesn't fit your vision for the template, but I thought I'd throw it out there.

Cameron

**Comment by h-enk** [comment]
@cameronelliott, thanks for bringing this up.

> Have you considered adding the newsletter form to the main Doks template?

No, I haven't — yet. It doesn't have my highest prio. However, I'm considering writing a guide on adding the [Buttondown](https://buttondown.email/) setup used by getdoks.org (rough draft already available).

What I like about Buttondown, is that it's privacy-friendly + GDPR-compliant, and priced friendly (and run by Justin only). Not hating Google (on the contrary), but I'd like to stay away from Google for privacy reasons.

This is roughly how I set it up for getdoks.org:

1. Add [newletter](https://github.com/h-enk/getdoks.org/blob/master/layouts/partials/sidebar/newsletter.html) form
2. Listen + process form submission (see also lines 36..64 of [app.js](https://github.com/h-enk/getdoks.org/blob/master/assets/js/app.js#L34-L64))
3. Submit to Netlify (as backup /fail-safe)
4. Use a [serverless function](https://github.com/h-enk/getdoks.org/blob/master/functions/submission-created.js) + the Buttondown API for further processing.

So, that's a possibility of doing it (and that I like /would recommend). Would you (and anyone else) be interested in a guide on the [Buttondown](https://buttondown.email/) setup used by getdoks.org?

**Reply by cameronelliott** [reply]
Henk, I didn't know about Buttondown, but it looks really cool, and I love his copy, so I am definitely going to keep it in mind. I suppose I am still exploring options for my newletter and email capture,  so I'd say there is a fair chance I may go a different route. I don't think the guide is a priority for me right now. Thank you though.

**Comment by Alrodr** [comment]
Hello again! I would like to retake this discussion and know if this guide on the [Buttondown](https://buttondown.email/) setup used by getdoks.org  it still stands? Thank you very much
